sites/docs/src/content/docs/tutorials/tests_and_test_data/components/03_testing_modules.md

```diff
@@ -0,0 +1,668 @@
+---
+title: "Testing nf-core/modules"
+subtitle: Writing comprehensive tests for nf-core modules with nf-test
+weight: 30
+---
+
+## Overview
+
+The nf-test framework in the nf-core ecosystem enables comprehensive testing for processes, workflows, and pipelines.
+
+This chapter covers the fundamentals of nf-core module testing, from basic syntax to advanced scenarios involving chained modules.
+
+## Understanding snapshots
+
+**What are snapshots?**
+
+Snapshots are nf-test's way of capturing and validating the expected outputs of your tests. When you run a test for the first time, nf-test creates a `.snap` file containing checksums, file names, and other metadata about that describe how a module's output should look like.
+
+**How snapshot matching works:**
+
+1. **First run**: nf-test generates a snapshot file (e.g., `main.nf.test.snap`) containing the expected outputs
+2. **Subsequent runs**: nf-test compares current outputs against the stored snapshot
+3. **Pass/Fail**: Test passes if outputs match the snapshot, fails if they differ
+
+**Why snapshots are useful:**
+
+- **Automated validation**: No need to manually check every output file
+- **Regression detection**: Automatically catch when module behavior changes
+- **Comprehensive checking**: Validates file content, structure, and metadata
+
+## Your first simple test walkthrough with an example
+
+Let's walk through creating a basic test step-by-step using the `cat` module as an example:
+
+### Step 1: Examine the module
+
+First, let's look at a simple module (`modules/cat/main.nf`):
+
+```groovy
+process CAT {
+    input:
+    tuple val(meta), path(file_in)
+
+    output:
+    tuple val(meta), path("${prefix}"), emit: file_out
+    path "versions.yml"               , emit: versions
+
+    script:
+    prefix = "${meta.id}.txt"
+    """
+    cat ${file_in} > ${prefix}
+    """
+}
+```
+
+This process simply concatenates input files and outputs a single file.
+
+### Step 2: Write the test
+
+Create `modules/cat/tests/main.nf.test`:
+
+```groovy
+nextflow_process {
+    name "Test Process CAT"
+    script "../main.nf"
+    process "CAT"
+
+    test("simple file concatenation") {
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'test' ], // meta map with sample identifier
+                    file('test_file.txt')  // input file
+                ]
+                """
+            }
+        }
+
+        then {
+            assert process.success               // Check process completed successfully
+            assert snapshot(process.out).match() // Validate outputs match expected snapshot
```

Comment:
```suggestion
            assert snapshot(sanitizeOutput(process.out)).match() // Validate outputs match expected snapshot
```
Might be worth adding `sanitizeOutput` here in the docs too as it will create easier to read snapshots